Add support for SDBus interfaces to launch and stop batch server

cmcantalupo opened the pull request:
- Relates to #1973
- Fixes #1988

bgeltz committed:
Add SDBus wrappers for <open|close>_container

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add SDBusMessage method to append geopm_request_s

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Implement ServiceProxyImp::platform_start_batch

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Fix mocks for missing methods

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup issues discovered during integration testing

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add SDBus method to call a function that takes one integer

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Move geopm_request_s into PlatformIO.hpp

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add unit test for ServiceProxy start_batch() and stop_batch()

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Change append_request_s() to apppend_request()

- We reserver _s for structure members, however this is a function.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add missing doxygen for new call_method() interface

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fix file path reporting in Exception error

- Not all files are in geopm/src/ anymore.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Some clean up for POSIXSignal mostly avoiding the "set" variable name

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo commented on service/src/POSIXSignal.hpp:
```diff
@@ -78,6 +78,13 @@ namespace geopm
             /// @return m_info_s reduced information set from siginfo_t struct
             virtual m_info_s reduce_info(const siginfo_t &info) const = 0;
 
+            // @brief Use stat() to get the get the uid of /proc/<PID>
+            // virtual int pid_to_uid(int pid);
+
+            // @brief Use stat() to get the get the gid of /proc/<PID>
+            // virtual int pid_to_gid(int pid);
+
+
```
```suggestion
```

cmcantalupo committed:
Remove commented code

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

Krebrov001 commented on service/src/POSIXSignal.hpp:
```diff
@@ -133,15 +135,16 @@ namespace geopm
             /// @remark See documentation for sigprocmask(2) about parameters
             ///         and return value.
-            virtual void sig_proc_mask(int how, const sigset_t *set, sigset_t *oldset) const = 0;
+            virtual void sig_proc_mask(int how, const sigset_t *sigset, sigset_t *oldset) const = 0;
```
```cpp
virtual void sig_proc_mask(int how, const sigset_t *sigset,
                           sigset_t *oldset) const = 0;
```

Krebrov001 commented on service/test/ServiceIOGroupTest.cpp:
```diff
@@ -37,6 +37,7 @@
 #include "geopm_topo.h"
 
 #include "geopm/Helper.hpp"
+#include "geopm/PlatformIO.hpp"
```
Why do we need this header for, there are no other changes to the file which would indicate need to include this header in the file.

Krebrov001 reviewed (changes requested):
Please implement these changes.

cmcantalupo committed:
Revert change to ServiceIOGroupTest.cpp that was accidentally committed

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Whitespace change to POSIXSignal.hpp

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
This change was actually required because of use of geopm_request_s

Revert "Revert change to ServiceIOGroupTest.cpp that was accidentally committed"

This reverts commit b9cfb317fca1f6fb72836355b6427b6f0dda4372.

Krebrov001 reviewed (approved):
cool thanks
